src/codesyncd/alert_utils.ts

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs';
+import yaml from 'js-yaml';
+import vscode from 'vscode';
+import { API_ROUTES, NOTIFICATION } from "../constants";
+import { viewDashboardHandler } from '../handlers/commands_handler';
+import { IRepoInfo, IUserProfile } from "../interface";
+import { CodeSyncLogger } from '../logger';
+import { generateSettings } from "../settings";
+import { getTeamActivity } from "../utils/api_utils";
+import { getActiveUsers, readYML } from "../utils/common";
+
+
+export class Alerts {
+	// In case we make it congifurable from user input in future, structure will help then.
+	CONFIG = {
+		TEAM_ACTIVITY : {
+			key: "team_activity",
+			showAt: {
+				hour: 16,
+				minutes: 30
+			},
+			showAfter: 24 * 60 * 60 * 1000, // 24 hours
+			api: API_ROUTES.TEAM_ACTIVITY
+		}
+	} 
+
+	settings: any;
+	nowHour: number;
+    nowMinutes: number;
+	nowTimestamp: number;
+	alertsData: any;
+	activeUser: IUserProfile;
+
+	constructor() {
+		const now = new Date();
+		this.nowHour = now.getHours();
+		this.nowMinutes = now.getMinutes();
+		this.nowTimestamp = new Date().getTime();
+		this.settings = generateSettings();
+		this.alertsData = readYML(this.settings.ALERTS);
+
+		this.activeUser = getActiveUsers()[0];
+	}
+
+	checkActivityAlerts = async () => {
+		if (!this.activeUser) return;
+		const accessToken = this.activeUser.access_token;
+		await this.checkTeamAlert(accessToken);
+	};
+
+	checkTeamAlert = async (accessToken: string) => {
+		/* 
+		Notify the user about recent activity within teams, daily at 4:30pm.
+		We need to check activity within past 24 hours i.e from 4:30PM yesterday to 4:30PM today.
+		Edge case:
+		- If user is not online at 4:30pm and becomes available at any time later (before 4:30PM next day);
+			- If there was an activity before 4:30PM, it will show an alert once and will not show alert
+			  again before 4:30PM of next day.
+			- If there was an activity after 4:30PM, it will not show the alert, rather will show the alert next day.
+
+		TODO: Show alert in all open instances of the IDE by keeping track in state variable. 
+		*/
+		// Check when last alert was shown to the user
+		const lastShownAT = this.alertsData[this.CONFIG.TEAM_ACTIVITY.key];
+		const canShowAlert = Boolean(lastShownAT && Math.abs(this.nowTimestamp - lastShownAT.getTime()) > this.CONFIG.TEAM_ACTIVITY.showAfter);
+		if (!lastShownAT) {
+			// show alert 
+			return await this.showTeamActivityAlert(accessToken);
+		}
+		const alertH = this.CONFIG.TEAM_ACTIVITY.showAt.hour;
+		const alertM = this.CONFIG.TEAM_ACTIVITY.showAt.minutes;
+		if (!canShowAlert && !(this.nowHour == alertH && this.nowMinutes >= alertM || this.nowHour > alertH)) return;
```

Comment:
@Basit-qc Shouldn't this be an `or` operation? i.e.
```
if (!canShowAlert ||!(this.nowHour == alertH && this.nowMinutes >= alertM || this.nowHour > alertH)) return;
```

Reply:
Let me know when you are back to discuss further.

Reply:
@saleem-latif I believe this has changed. Kindly check in master branch. Thanks